python/tests/xir/test_program.py

```diff
@@ -23,6 +24,30 @@ def program():
     return XIRProgram()
 
 
+# pylint: disable=protected-access
+def make_program(
+    called_functions: Set[str],
+    declarations: Dict[str, List[Declaration]],
+    gates: Dict[str, Dict[str, Any]],
+    includes: List[str],
+    operators: Dict[str, Dict[str, Any]],
+    options: Dict[str, Any],
+    statements: Set[str],
+    variables: List[str],
+):
+    """Returns an XIR program with the given attributes."""
+    program = XIRProgram()
+    program._called_functions = called_functions
+    program._declarations = declarations
+    program._gates = gates
+    program._includes = includes
+    program._operators = operators
+    program._options = options
+    program._statements = statements
+    program._variables = variables
```

Comment:
Yes. I chose to modify the private attributes directly because the code which used the public mutators felt more verbose than necessary (~3 times as many lines); I like to think of this function more as a private constructor of `XIRProgram`.

Do you think we should use the public mutators anyway?

Reply:
I'm honestly happy with this, I'm just trying to think what's best practice. 😄